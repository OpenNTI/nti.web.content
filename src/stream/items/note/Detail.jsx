import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import {LuckyCharms} from '@nti/web-commons';
import {Panel as Body} from '@nti/web-modeled-content';
import { Context } from '@nti/web-discussions';
import { LinkTo } from '@nti/web-routing';
import { getService } from '@nti/web-client';

import {ItemLinks, Avatar} from '../components';

import Meta from './Meta';

export default class NoteDetils extends React.Component {
	static propTypes = {
		item: PropTypes.shape({
			body: PropTypes.any.isRequired,
			isReply: PropTypes.func.isRequired,
			creator: PropTypes.string.isRequired,
			inReplyTo: PropTypes.string,
			getID: PropTypes.func.isRequired,
			title: PropTypes.string
		}).isRequired
	}

	state = {
		reply: null
	}

	componentDidMount () {
		this.resolveReply();
	}

	componentDidUpdate (prevProps) {
		if (prevProps.item.getID() !== this.props.item.getID()) {
			this.resolveReply();
		}
	}

	resolveReply = async () => {
		const { item } = this.props;

		if (!item.isReply()) { return; }

		const service = await getService();
		const reply = await service.getObject(item.inReplyTo);
		this.setState({ reply });
	}

	render () {
		const {item} = this.props;
		const {reply} = this.state;
		const {body, creator, placeholder} = item;
		const isReply = item.isReply();

		return (
			<div className={cx('stream-note-details', {'is-reply': isReply})}>
				<LinkTo.Object object={item} context="stream-context">
					<div className="context">
						<Context item={item} />
					</div>
				</LinkTo.Object>
				<div className="item">
					<LuckyCharms item={item} />
					<div className="title-container">
						<Avatar creator={creator} />
						<Meta item={item} reply={reply} />
					</div>
					{!placeholder && (
						<Body body={body} />
					)}
					<ItemLinks item={item} />
				</div>
			</div>
		);
	}
}
