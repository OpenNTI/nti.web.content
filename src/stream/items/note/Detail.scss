@import '~@nti/style-common';

.nti-content-stream-note-details {
	position: relative;
	padding: 20px;

	.title-container {
		margin-right: 45px;
		display: flex;
		flex-direction: row;

		.avatar-container{
			flex: 0 0 auto;
			width: 3.75rem;
			height: 3.75rem;
			margin-right: 20px;

			.avatar {
				width: 100%;
				height: 100%;
			}
		}

		.meta {
			flex: 1 1 auto;

			.title {
				margin: 0;
				font: normal 300 1.5rem/1.3 $body-font-family;
				color: $primary-grey;
			}
		}

		.reply-name-wrapper, .name-wrapper {
			list-style: none;
			padding: 0;
			margin: 10px 0 0;
			font: normal 600 0.75rem/1.3 $body-font-family;

			li {
				display: inline-block;

				&:not(:first-of-type) {
					@extend %middot-before;
				}
			}
		}
	}

	a[rel=author] {
		display: inline-block;
		max-width: 12.5rem;
		overflow: hidden;
		white-space: nowrap;
		text-overflow: ellipsis;
		vertical-align: bottom;
	}

	.note-content {
		padding: 20px 0;

		.whiteboard {
			max-width: 200px;
			margin: 0.1875rem;
			margin-bottom: 0.5rem;
			border: 1px solid $border-grey-alt;
			box-shadow: 0.125rem 0.1875rem 0.0625rem 0 rgba(0,0,0,0.13);
			border-radius: 1px;//iOS fix for box-shadow
		}
	}
}
