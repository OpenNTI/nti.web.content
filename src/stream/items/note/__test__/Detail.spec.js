import React from 'react';
import renderer from 'react-test-renderer';
import { TestUtils } from '@nti/web-client';

import Detail from '../Detail';

const { tearDownTestClient, setupTestClient } = TestUtils;

const mockService = {
	resolveEntity: (user) => {
		if (user === 'User') {
			return Promise.resolve({ displayName: 'Tim Jones' });
		} else if (user === 'user2') {
			return Promise.resolve({ displayName: 'Emily Harry' });
		}
	},
	getObject: () => {
		return Promise.resolve({
			creator: 'user2'
		});
	}
};

const onBefore = () => {
	jest.useFakeTimers();
	setupTestClient(mockService);
};

const onAfter = () => {
	tearDownTestClient();
};

const flushPromises = () => new Promise(resolve => setImmediate(resolve));

/* eslint-env jest */
describe('Note Details', () => {
	beforeEach(onBefore);
	afterEach(onAfter);

	test('General: (Liked, Favorited, Tim Jones, A Day Ago)', async () => {
		const item = {
			body: ['This is the body.'],
			isReply: () => false,
			title: 'Note Title',
			creator: 'User',
			getCreatedTime: () => new Date() - (1000 * 60 * 60 * 24),
			isTopLevel: () => true,
			LikeCount: 1,
			Links: ['unlike','unfavorite'],
			hasLink: function (link) { return this.Links.includes(link); },
			MimeType: 'application/vnd.nextthought.note',
			containerId: 'tag:nextthought.com,2011-10:NTI-HTML-8211584329684854468',
			getContextData: () => Promise.resolve(),
			getID: () => 'tag:nextthought.com,2011-10:user2-OID-0x05819c:5573657273:aZT2yH3CDrT',
			inReplyTo: 'tag:nextthought.com,2011-10:user2-OID-0x05763d:5573657273:PRXB6WxX47U'
		};

		const noteDetail = renderer.create(<Detail item={item} />);

		jest.runAllTimers();
		await flushPromises();

		const tree = noteDetail.toJSON();
		expect(tree).toMatchSnapshot();
	});

	test('General: (Tim Jones, A 2 Days Ago)', async () => {
		const item = {
			body: ['This is the body.'],
			isReply: () => false,
			title: 'Note Title',
			creator: 'User',
			getCreatedTime: () => new Date() - (1000 * 60 * 60 * 48),
			isTopLevel: () => true,
			LikeCount: 0,
			Links: [],
			hasLink: function (link) { return this.Links.includes(link); },
			MimeType: 'application/vnd.nextthought.note',
			containerId: 'tag:nextthought.com,2011-10:NTI-HTML-8211584329684854468',
			getContextData: () => Promise.resolve(),
			getID: () => 'tag:nextthought.com,2011-10:user2-OID-0x05819c:5573657273:aZT2yH3CDrT'
		};

		const noteDetail = renderer.create(<Detail item={item} />);

		jest.runAllTimers();
		await flushPromises();

		const tree = noteDetail.toJSON();
		expect(tree).toMatchSnapshot();
	});

	test('Reply', async () => {
		const item = {
			body: ['This is the body.'],
			isReply: () => true,
			title: 'Note Title',
			creator: 'User',
			getCreatedTime: () => new Date() - (1000 * 60 * 60 * 48),
			isTopLevel: () => true,
			LikeCount: 0,
			Links: [],
			hasLink: function (link) { return this.Links.includes(link); },
			MimeType: 'application/vnd.nextthought.note',
			containerId: 'tag:nextthought.com,2011-10:NTI-HTML-8211584329684854468',
			getContextData: () => Promise.resolve(),
			getID: () => 'tag:nextthought.com,2011-10:user2-OID-0x05819c:5573657273:aZT2yH3CDrT'
		};

		const noteDetail = renderer.create(<Detail item={item} />);

		jest.runAllTimers();
		await flushPromises();

		const tree = noteDetail.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
