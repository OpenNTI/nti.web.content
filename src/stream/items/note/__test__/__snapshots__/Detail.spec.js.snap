// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note Details General: (Liked, Favorited, Tim Jones, A Day Ago) 1`] = `
<div
  className="nti-content-stream-note-details"
>
  <div
    className="nti-link-to-path"
    to="#"
  >
    <div
      className="context"
    >
      <div
        className="discussion-context-view"
      />
    </div>
  </div>
  <div
    className="detail"
  >
    <div
      className="title-container"
    >
      <div
        className="charms"
      >
        <a
          className="like active"
          href="#"
          onClick={[Function]}
        >
          1
        </a>
        <a
          className="favorite active"
          href="#"
          onClick={[Function]}
        />
      </div>
      <div
        className="avatar-container"
      >
        <div
          className="nti-link-to-path"
          to="#"
        >
          <img
            alt="Avatar for Tim Jones"
            className="avatar avatar-color-6"
            src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAP8AAAD/CAMAAAAJ1vD4AAACAVBMVEXN1trFz9O4wsistr2gq7OWoaqMl6GBjZh5hZF+iZWCjpmHkp2LlqGPm6SUn6mZpK2eqbGkr7eqtbyxu8K3wce9x8zFztPL1Nm8xcubpa59iZR4hJB6hpKDj5qMmKKXoqukrrayvMLDzdKirLR7h5Kbpq+2wMbL1NjM1dm0vsV9iZV/i5ahrLTGz9TI0dalr7d8iJPK09eos7p/ipaDjpmzvcPCy9CTnqitt76fqbKRnKbBys+KlqB+ipa0vsSqtLvDzNGOmaOUn6iQm6WVoKmapK2msLjJ0te6xMqKlaCGkpzK09iSnad8iJSZpKynsbm6xMmapa7H0NSSnaauuL+JlZ+8xsujrbWRnKV6hpGps7udqLG+yM1+ipW/yM6Ik56NmKLAyc65w8mxu8GwusGosrmirbWfqrKbpq6dqLCeqLGmsbizvcTH0dWAjJeNmaOKlZ+Yo6yps7q+x82FkZuFkJulsLeosrqvucDEztK2v8a4wce3wMe7xMq1v8WEj5p7h5OyvMPCzNGstr7G0NSPmqSwusCrtr2rtbzAys+Hk52qtLy9xsyBjJicpq+jrra/yc7EzdLJ0taptLuLlqCIlJ6cp7CEkJvI0dWTnqe5wsicp6+EkJqhq7O7xcvH0NWJlJ+GkZyCjZi7xcqAi5fBytDAyc+Ll6GnsrnBy9COmqMAgrVeAAAF/UlEQVR4AezBBQHAIAAAsLsr1j8qDUiwrQMAAAAAAAAAAACAJvphnOZl3fbjvO7n/f4QUy61vXtvi6u6wgD+Gm4nQwRbQpB5h2SSGOZoYJgJZrhEcCYg11AzCc04hgS1SDEXI2osppGMCCkmxUpMb6b31l78lH36F7alBph9Fmuf3d83eJ9n7/WcfVsHDmh6+rvN1VFuKtZy8FD88JHwZt9/tPoZPlbkWGsCoeM/W/Uct+p4WztCJdlRw21JxdMIDf9EJ7ft+ZMIicw+7kjXAYRBQ5Q7tLcb1vN7WIbeA7DbqRdYlr5+2OzFFMtV6cNW2RwNOJ2Gnfr7aMRAAjaK05RB2Md/ieYMwTbDvTTIG4FdkqM0aiwBm4znaFgcFvHP0LSJJ2CP79G8ZljjZQbAy8ASZ2MMQh52OBdhIGJHYIXzDEik24d+kwzO9y9Au3N9DJBX8KHbMQbrlWFolikyYLkEFHuJgauCXtkYg/cq1LpIAbEpKJW4RAk5KHWSMi5DpzxlFKDTNGW8BpWGPcqY8KHR65SSgUb7KeUNaPQDSpmBRico5U1oNEspPdDoh5TSDI3mKKUKGtVTyiA0qqKUXmj0//xS6hyf/1WO1/96aPQWpcxBo0pKuQKN2iilEhpdpZTvQKNuSqmARtco5TA0GqGUPdDoFKW8Do2uU0o7VCpShudDpRbKeBs63aCMd6DTPGV0Qac4ZbwFnWYo413o9B5l9EOntOf2/Qe8TwleE5QapISb0KqDEs5Aq1ZKaINWSY8CZqDWmNPlH6ingCTU+oDBi0GvH1HAOFRKLzSkZykg+2H3LejTS749SgFdMfJlh3c/dW4BVVBUBsoUKOrHUOYiRfVDmUqKuu34+P/I8fqXhTKLlOQloEyGku5AG79EQb1Q5wUKOgp15ijoY6gzSUHtUGeJcm5CoecpJg+F8hTTAIUOU0osCYXSKQoZgEp1FPKJ2/cfvGWoNFyiiBUodYgi7kKpPZRwPOH2+Wcl1PoJg+ctQa1EjdsNoLDKoBU/hWL+PQbsIlR7ushA3RmGbvcZpOgUtJupzjMYLacHfgobrDEQ87BEFwOxCksMMRCNsMQbDELUhyWaUgzAeVhjgAF40/EW8FOwximaF4FFpmncZ7DIz2jcLVjkc5p2KQGbjNGwPKwyK9DzwKUJEPFhl2oaVYBlJmmStw7LpDtpUB2sU6BBI7DOconG7IOFemjMs7DQcpSGjMJKBRrSCis9OE4jBmCpL2hC7DIs5T+kAW2w1lSRZXt/HPZqk1/4qTJ+T/68W5XPSyzLOwnYbYHleG4Jtuvhznm3IE/R35DvIwTSN7hDswiF4Z9zR36BkMjLv/FWpc7x/CuO5691PP+E2/mv0+38jY7nP+x4/iHH8+cdz59zNn/7KzdWVlZSzubPsRzNq1fj7bDYNZbtl7+CvVZYvnlYK0sDvEbYaogmDMJWozTBy8JOH9KMWdipgma0wE6/piFfQp6mRzCPYKP9NKUGNnqKxpyFhWppTCWsM3KF5tTCLh9didCoLOxxue03NO0ELJEdmmYQxn77O6j3+9WDDE7NXGsCeqVnzsQYsGjVzANo5N+ej1JEbOD+OpS58NQaJf2hcAFqZOJ/pLw/zS0msPvOvVjN3ZI6PbmM3ZS+21vkrvIODn2K3eG3zk9Qg5s9X/qQ1tizRj36DjU8gJwjH9RSm9iTX2QhIdmd86jTdNuffQSq6dpgiZr9pf7jNIJy9rNO6vdM7+o6zPtq9g4F6PxAXH51Hy1T07w4bmphd75IG6W6PmlHufbMR2kv72HHX7Fz4xX3aL2/TSawI9ePdjIUOivbsW3JQpShUXqUxbakOy4xVIr1B7B1dyMMnVQ8ja3JPMlQiixgK/5eYli99g88zvrXDLHSu/h2t/sYbgPL+BZXPYZdpBH/i19PB0wsYnOJOjqhOInN+IN0hNeATTyiM2KL+C/ddMjeJfyHr1K0l4G+Gjm6JY5/c5KOmXgC3/SQrjmGb3iPzkklsaFA9yxgw9d0zyFsWKN7xrAhxi0r7jXD4y6bwL/8Ezfe7zwt0pvSAAAAAElFTkSuQmCC"
          />
        </div>
      </div>
      <div
        className="meta"
      >
        <h1
          className="title"
        >
          Note Title
        </h1>
        <ul
          className="name-wrapper"
        >
          <li>
            <div
              className="nti-link-to-path"
              to="#"
            >
              <address
                className="username"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "Posted by <a rel=\\"author\\" class=\\"username\\">Tim Jones</a>",
                  }
                }
                rel="author"
              />
            </div>
          </li>
          <li>
            1 Day ago
          </li>
        </ul>
      </div>
    </div>
    <div
      className="note-content"
    >
      <div
        className="modeled-content"
      >
        <div>
          This is the body.
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Note Details General: (Tim Jones, A 2 Days Ago) 1`] = `
<div
  className="nti-content-stream-note-details"
>
  <div
    className="nti-link-to-path"
    to="#"
  >
    <div
      className="context"
    >
      <div
        className="discussion-context-view"
      />
    </div>
  </div>
  <div
    className="detail"
  >
    <div
      className="title-container"
    >
      <div
        className="charms"
      >
        <a
          className="like"
          href="#"
          onClick={[Function]}
        >
          
        </a>
        <a
          className="favorite"
          href="#"
          onClick={[Function]}
        />
      </div>
      <div
        className="avatar-container"
      >
        <div
          className="nti-link-to-path"
          to="#"
        >
          <img
            alt="Avatar for Tim Jones"
            className="avatar avatar-color-6"
            src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAP8AAAD/CAMAAAAJ1vD4AAACAVBMVEXN1trFz9O4wsistr2gq7OWoaqMl6GBjZh5hZF+iZWCjpmHkp2LlqGPm6SUn6mZpK2eqbGkr7eqtbyxu8K3wce9x8zFztPL1Nm8xcubpa59iZR4hJB6hpKDj5qMmKKXoqukrrayvMLDzdKirLR7h5Kbpq+2wMbL1NjM1dm0vsV9iZV/i5ahrLTGz9TI0dalr7d8iJPK09eos7p/ipaDjpmzvcPCy9CTnqitt76fqbKRnKbBys+KlqB+ipa0vsSqtLvDzNGOmaOUn6iQm6WVoKmapK2msLjJ0te6xMqKlaCGkpzK09iSnad8iJSZpKynsbm6xMmapa7H0NSSnaauuL+JlZ+8xsujrbWRnKV6hpGps7udqLG+yM1+ipW/yM6Ik56NmKLAyc65w8mxu8GwusGosrmirbWfqrKbpq6dqLCeqLGmsbizvcTH0dWAjJeNmaOKlZ+Yo6yps7q+x82FkZuFkJulsLeosrqvucDEztK2v8a4wce3wMe7xMq1v8WEj5p7h5OyvMPCzNGstr7G0NSPmqSwusCrtr2rtbzAys+Hk52qtLy9xsyBjJicpq+jrra/yc7EzdLJ0taptLuLlqCIlJ6cp7CEkJvI0dWTnqe5wsicp6+EkJqhq7O7xcvH0NWJlJ+GkZyCjZi7xcqAi5fBytDAyc+Ll6GnsrnBy9COmqMAgrVeAAAF/UlEQVR4AezBBQHAIAAAsLsr1j8qDUiwrQMAAAAAAAAAAACAJvphnOZl3fbjvO7n/f4QUy61vXtvi6u6wgD+Gm4nQwRbQpB5h2SSGOZoYJgJZrhEcCYg11AzCc04hgS1SDEXI2osppGMCCkmxUpMb6b31l78lH36F7alBph9Fmuf3d83eJ9n7/WcfVsHDmh6+rvN1VFuKtZy8FD88JHwZt9/tPoZPlbkWGsCoeM/W/Uct+p4WztCJdlRw21JxdMIDf9EJ7ft+ZMIicw+7kjXAYRBQ5Q7tLcb1vN7WIbeA7DbqRdYlr5+2OzFFMtV6cNW2RwNOJ2Gnfr7aMRAAjaK05RB2Md/ieYMwTbDvTTIG4FdkqM0aiwBm4znaFgcFvHP0LSJJ2CP79G8ZljjZQbAy8ASZ2MMQh52OBdhIGJHYIXzDEik24d+kwzO9y9Au3N9DJBX8KHbMQbrlWFolikyYLkEFHuJgauCXtkYg/cq1LpIAbEpKJW4RAk5KHWSMi5DpzxlFKDTNGW8BpWGPcqY8KHR65SSgUb7KeUNaPQDSpmBRico5U1oNEspPdDoh5TSDI3mKKUKGtVTyiA0qqKUXmj0//xS6hyf/1WO1/96aPQWpcxBo0pKuQKN2iilEhpdpZTvQKNuSqmARtco5TA0GqGUPdDoFKW8Do2uU0o7VCpShudDpRbKeBs63aCMd6DTPGV0Qac4ZbwFnWYo413o9B5l9EOntOf2/Qe8TwleE5QapISb0KqDEs5Aq1ZKaINWSY8CZqDWmNPlH6ingCTU+oDBi0GvH1HAOFRKLzSkZykg+2H3LejTS749SgFdMfJlh3c/dW4BVVBUBsoUKOrHUOYiRfVDmUqKuu34+P/I8fqXhTKLlOQloEyGku5AG79EQb1Q5wUKOgp15ijoY6gzSUHtUGeJcm5CoecpJg+F8hTTAIUOU0osCYXSKQoZgEp1FPKJ2/cfvGWoNFyiiBUodYgi7kKpPZRwPOH2+Wcl1PoJg+ctQa1EjdsNoLDKoBU/hWL+PQbsIlR7ushA3RmGbvcZpOgUtJupzjMYLacHfgobrDEQ87BEFwOxCksMMRCNsMQbDELUhyWaUgzAeVhjgAF40/EW8FOwximaF4FFpmncZ7DIz2jcLVjkc5p2KQGbjNGwPKwyK9DzwKUJEPFhl2oaVYBlJmmStw7LpDtpUB2sU6BBI7DOconG7IOFemjMs7DQcpSGjMJKBRrSCis9OE4jBmCpL2hC7DIs5T+kAW2w1lSRZXt/HPZqk1/4qTJ+T/68W5XPSyzLOwnYbYHleG4Jtuvhznm3IE/R35DvIwTSN7hDswiF4Z9zR36BkMjLv/FWpc7x/CuO5691PP+E2/mv0+38jY7nP+x4/iHH8+cdz59zNn/7KzdWVlZSzubPsRzNq1fj7bDYNZbtl7+CvVZYvnlYK0sDvEbYaogmDMJWozTBy8JOH9KMWdipgma0wE6/piFfQp6mRzCPYKP9NKUGNnqKxpyFhWppTCWsM3KF5tTCLh9didCoLOxxue03NO0ELJEdmmYQxn77O6j3+9WDDE7NXGsCeqVnzsQYsGjVzANo5N+ej1JEbOD+OpS58NQaJf2hcAFqZOJ/pLw/zS0msPvOvVjN3ZI6PbmM3ZS+21vkrvIODn2K3eG3zk9Qg5s9X/qQ1tizRj36DjU8gJwjH9RSm9iTX2QhIdmd86jTdNuffQSq6dpgiZr9pf7jNIJy9rNO6vdM7+o6zPtq9g4F6PxAXH51Hy1T07w4bmphd75IG6W6PmlHufbMR2kv72HHX7Fz4xX3aL2/TSawI9ePdjIUOivbsW3JQpShUXqUxbakOy4xVIr1B7B1dyMMnVQ8ja3JPMlQiixgK/5eYli99g88zvrXDLHSu/h2t/sYbgPL+BZXPYZdpBH/i19PB0wsYnOJOjqhOInN+IN0hNeATTyiM2KL+C/ddMjeJfyHr1K0l4G+Gjm6JY5/c5KOmXgC3/SQrjmGb3iPzkklsaFA9yxgw9d0zyFsWKN7xrAhxi0r7jXD4y6bwL/8Ezfe7zwt0pvSAAAAAElFTkSuQmCC"
          />
        </div>
      </div>
      <div
        className="meta"
      >
        <h1
          className="title"
        >
          Note Title
        </h1>
        <ul
          className="name-wrapper"
        >
          <li>
            <div
              className="nti-link-to-path"
              to="#"
            >
              <address
                className="username"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "Posted by <a rel=\\"author\\" class=\\"username\\">Tim Jones</a>",
                  }
                }
                rel="author"
              />
            </div>
          </li>
          <li>
            2 Days ago
          </li>
        </ul>
      </div>
    </div>
    <div
      className="note-content"
    >
      <div
        className="modeled-content"
      >
        <div>
          This is the body.
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Note Details Reply 1`] = `
<div
  className="nti-content-stream-note-details is-reply"
>
  <div
    className="nti-link-to-path"
    to="#"
  >
    <div
      className="context"
    >
      <div
        className="discussion-context-view"
      />
    </div>
  </div>
  <div
    className="detail"
  >
    <div
      className="title-container"
    >
      <div
        className="charms"
      >
        <a
          className="like"
          href="#"
          onClick={[Function]}
        >
          
        </a>
        <a
          className="favorite"
          href="#"
          onClick={[Function]}
        />
      </div>
      <div
        className="avatar-container"
      >
        <div
          className="nti-link-to-path"
          to="#"
        >
          <img
            alt="Avatar for Tim Jones"
            className="avatar avatar-color-6"
            src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAP8AAAD/CAMAAAAJ1vD4AAACAVBMVEXN1trFz9O4wsistr2gq7OWoaqMl6GBjZh5hZF+iZWCjpmHkp2LlqGPm6SUn6mZpK2eqbGkr7eqtbyxu8K3wce9x8zFztPL1Nm8xcubpa59iZR4hJB6hpKDj5qMmKKXoqukrrayvMLDzdKirLR7h5Kbpq+2wMbL1NjM1dm0vsV9iZV/i5ahrLTGz9TI0dalr7d8iJPK09eos7p/ipaDjpmzvcPCy9CTnqitt76fqbKRnKbBys+KlqB+ipa0vsSqtLvDzNGOmaOUn6iQm6WVoKmapK2msLjJ0te6xMqKlaCGkpzK09iSnad8iJSZpKynsbm6xMmapa7H0NSSnaauuL+JlZ+8xsujrbWRnKV6hpGps7udqLG+yM1+ipW/yM6Ik56NmKLAyc65w8mxu8GwusGosrmirbWfqrKbpq6dqLCeqLGmsbizvcTH0dWAjJeNmaOKlZ+Yo6yps7q+x82FkZuFkJulsLeosrqvucDEztK2v8a4wce3wMe7xMq1v8WEj5p7h5OyvMPCzNGstr7G0NSPmqSwusCrtr2rtbzAys+Hk52qtLy9xsyBjJicpq+jrra/yc7EzdLJ0taptLuLlqCIlJ6cp7CEkJvI0dWTnqe5wsicp6+EkJqhq7O7xcvH0NWJlJ+GkZyCjZi7xcqAi5fBytDAyc+Ll6GnsrnBy9COmqMAgrVeAAAF/UlEQVR4AezBBQHAIAAAsLsr1j8qDUiwrQMAAAAAAAAAAACAJvphnOZl3fbjvO7n/f4QUy61vXtvi6u6wgD+Gm4nQwRbQpB5h2SSGOZoYJgJZrhEcCYg11AzCc04hgS1SDEXI2osppGMCCkmxUpMb6b31l78lH36F7alBph9Fmuf3d83eJ9n7/WcfVsHDmh6+rvN1VFuKtZy8FD88JHwZt9/tPoZPlbkWGsCoeM/W/Uct+p4WztCJdlRw21JxdMIDf9EJ7ft+ZMIicw+7kjXAYRBQ5Q7tLcb1vN7WIbeA7DbqRdYlr5+2OzFFMtV6cNW2RwNOJ2Gnfr7aMRAAjaK05RB2Md/ieYMwTbDvTTIG4FdkqM0aiwBm4znaFgcFvHP0LSJJ2CP79G8ZljjZQbAy8ASZ2MMQh52OBdhIGJHYIXzDEik24d+kwzO9y9Au3N9DJBX8KHbMQbrlWFolikyYLkEFHuJgauCXtkYg/cq1LpIAbEpKJW4RAk5KHWSMi5DpzxlFKDTNGW8BpWGPcqY8KHR65SSgUb7KeUNaPQDSpmBRico5U1oNEspPdDoh5TSDI3mKKUKGtVTyiA0qqKUXmj0//xS6hyf/1WO1/96aPQWpcxBo0pKuQKN2iilEhpdpZTvQKNuSqmARtco5TA0GqGUPdDoFKW8Do2uU0o7VCpShudDpRbKeBs63aCMd6DTPGV0Qac4ZbwFnWYo413o9B5l9EOntOf2/Qe8TwleE5QapISb0KqDEs5Aq1ZKaINWSY8CZqDWmNPlH6ingCTU+oDBi0GvH1HAOFRKLzSkZykg+2H3LejTS749SgFdMfJlh3c/dW4BVVBUBsoUKOrHUOYiRfVDmUqKuu34+P/I8fqXhTKLlOQloEyGku5AG79EQb1Q5wUKOgp15ijoY6gzSUHtUGeJcm5CoecpJg+F8hTTAIUOU0osCYXSKQoZgEp1FPKJ2/cfvGWoNFyiiBUodYgi7kKpPZRwPOH2+Wcl1PoJg+ctQa1EjdsNoLDKoBU/hWL+PQbsIlR7ushA3RmGbvcZpOgUtJupzjMYLacHfgobrDEQ87BEFwOxCksMMRCNsMQbDELUhyWaUgzAeVhjgAF40/EW8FOwximaF4FFpmncZ7DIz2jcLVjkc5p2KQGbjNGwPKwyK9DzwKUJEPFhl2oaVYBlJmmStw7LpDtpUB2sU6BBI7DOconG7IOFemjMs7DQcpSGjMJKBRrSCis9OE4jBmCpL2hC7DIs5T+kAW2w1lSRZXt/HPZqk1/4qTJ+T/68W5XPSyzLOwnYbYHleG4Jtuvhznm3IE/R35DvIwTSN7hDswiF4Z9zR36BkMjLv/FWpc7x/CuO5691PP+E2/mv0+38jY7nP+x4/iHH8+cdz59zNn/7KzdWVlZSzubPsRzNq1fj7bDYNZbtl7+CvVZYvnlYK0sDvEbYaogmDMJWozTBy8JOH9KMWdipgma0wE6/piFfQp6mRzCPYKP9NKUGNnqKxpyFhWppTCWsM3KF5tTCLh9didCoLOxxue03NO0ELJEdmmYQxn77O6j3+9WDDE7NXGsCeqVnzsQYsGjVzANo5N+ej1JEbOD+OpS58NQaJf2hcAFqZOJ/pLw/zS0msPvOvVjN3ZI6PbmM3ZS+21vkrvIODn2K3eG3zk9Qg5s9X/qQ1tizRj36DjU8gJwjH9RSm9iTX2QhIdmd86jTdNuffQSq6dpgiZr9pf7jNIJy9rNO6vdM7+o6zPtq9g4F6PxAXH51Hy1T07w4bmphd75IG6W6PmlHufbMR2kv72HHX7Fz4xX3aL2/TSawI9ePdjIUOivbsW3JQpShUXqUxbakOy4xVIr1B7B1dyMMnVQ8ja3JPMlQiixgK/5eYli99g88zvrXDLHSu/h2t/sYbgPL+BZXPYZdpBH/i19PB0wsYnOJOjqhOInN+IN0hNeATTyiM2KL+C/ddMjeJfyHr1K0l4G+Gjm6JY5/c5KOmXgC3/SQrjmGb3iPzkklsaFA9yxgw9d0zyFsWKN7xrAhxi0r7jXD4y6bwL/8Ezfe7zwt0pvSAAAAAElFTkSuQmCC"
          />
        </div>
      </div>
      <div
        className="meta"
      >
        <ul
          className="reply-name-wrapper"
        >
          <li>
            <div
              className="nti-link-to-path"
              to="#"
            >
              <span
                className="username"
                rel="author"
              >
                Tim Jones
              </span>
            </div>
            <span
              className="replied-to"
            >
               replied to 
            </span>
            <div
              className="nti-link-to-path"
              to="#"
            >
              <span
                className="username"
                rel="author"
              >
                Emily Harry
              </span>
            </div>
          </li>
          <li>
            2 Days ago
          </li>
        </ul>
      </div>
    </div>
    <div
      className="note-content"
    >
      <div
        className="modeled-content"
      >
        <div>
          This is the body.
        </div>
      </div>
    </div>
  </div>
</div>
`;
